### ISSUE TITLE
New Home page

### ISSUE THREAD

**Issue by sujato** [suttacentral/suttacentral#1629]
I have written a new home page from the ground up. It no longer relies on either polymer or static-styles.

- All styles are included. Prefer responsive units `em`, `%`, `vw`.
- It's just HTML, <s>with the exception of `<mwc-button>`.</s>
- Remove Vimeo, use native `<video>` element served from internet archive.
- Use new images, all simply Buddhist texts. Use `webm` with `jpg` fallback.
- Get rid of carousel
-  The current i18n implementation works, but I didn't change the file in the /templates folder. 

## To do

- [x]  Add `jpg` fallback images for iOS. (sujato)
- [x]  Adjust responsive styling. (sujato)
- [ ] Make quote randomizer (see below). (hongda)
- [x] Add ARIA metadata (Aminah)

### Quote randomizer

Currently we have a quotation carousel, which we use in two places:

1. Quotes from the suttas
2. Quotes from readers.

The quotation carousel was a bad idea, complicated and over-engineered. Also really bad for a11y.

Instead, simply serve a **different quote for each page refresh**. We can use the same sources as before.

As before, the button for the sutta quotes should link to the appropriate sutta. For the reader quotes, there is just a static link to Discourse.

**Comment by Aminah-SC** [comment]
As per my comment in Gitter, it's been a mighty, mighty long while since I've engaged with any a11y research and this coupled with the fact that I've never more than glanced at the SC app code base, means I'm not exactly overly qualified to treat this important task with the skilled care it deserves! I note this in particular, should any a11y experts with some free time on their hands happen to be perusing our issues. 

That said, by the "something better than nothing" principle, of course, I did want to take a look at this. The following is where my adventures have lead me so far. 

## Steps

- Checkout the user [experience of the updated homepage through NVDA*](https://drive.google.com/open?id=1_4xSe-kKuvJYOhmxDz2pEXxeR-2HSZYr)
- Explore the homepage via the accessibility feature in FF's inspector which highlights a bunch of issues with the page. Many (and the most significant) of them don't originate from `home-page.js `, but rather `sc-more-menu.js` and other js.
- Conclude it's not worth looking at ARIA at this stage (as things stand, the only obvious fix I think I could make would be to add proper descriptions to the "more" buttons). I'd suggest sorting out the js first and coming back to any necessary ARIA thereafter (just to note, the gist of the general guidance is to use ARIA sparingly / as a last resort as [ARIA was born as a fix to pre-HTML5 problems](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA))
- Find [W3's new latest guide to using ARIA](https://w3c.github.io/using-aria/) for closer review.  

## Some points raised by the FF inspector

- `<main>` is clickable element, I'm not sure why…? 

   > Clickable elements must be focusable and should have interactive semantics.

- This "clickable element" issue pops up a lot and is tied to some js that I can't unravel. 
   - As an aside, my guess would be that what ever's going on here, it's is also likely to be the reason why if you tab through the homepage you end up being taken to a whole bunch of "phantom focus points", a significant issue in terms of accessibility when tabbing is a key navigation method (see [the forth rule of Fight Club](https://w3c.github.io/using-aria/#fourth)).  
- The video player the Internet Archive uses appears to have some high-grade accessibility issues which in turn manifest on on SC as well ([here's a list of accessible AV players](https://www.digitala11y.com/accessible-jquery-html5-media-players/)).
    - In addition the `tabindex` attr. is currently set to `-1` Is this for a reason? ([More about the `tabindex`](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex)
- Other issues are flagged (such as contrast and focus styling), but for the minute it's probably best to leave it at this. 

I don't know if @firepick1 has anything to add?

-----
\* <sup>NVDA is currently the [leading screen reader](https://webaim.org/projects/screenreadersurvey8/#primary_div). Ideally, it would be good to test with JAWS, too, as it really, it shares "first place" with NVDA (as a total aside, it's a beautiful thing to see NVDA take the prime spot from JAWS so quickly; it's an awesome project!), however, the free version can only be used for a limited period and is an utter pain in the bum to test with. [Screen reader behaviour varies)(https://www.levelaccess.com/how-browsers-interact-with-screen-readers-and-where-aria-fits-in-the-mix/) from reader/browser combo to reader/browser combo which means that it's difficult to get things perfect, nevertheless, for what's being done here, its good enough. </sup>

**Comment by sujato** [comment]
I should have been more clear, for now we are looking solely at the content of the Home page, i.e. what is between the `<main>` tags.

I don't see phantom focus points when tabbing: tabbing goes through the buttons, which I think is correct, except it omits the video. (Note, I've changed the button markup for unrelated reasons)

> The video player the Internet Archive uses

It doesn't really work like that. The video is purely using the native player via the `<video>` element. IA is merely serving the file. that's why, if you view it in different browsers/platforms it appears quite different.

> the tabindex attr. is currently set to -1 Is this for a reason?

Nope, just a mistake, I've taken that out. Tabbing the video works fine in Chrome, but in FF it only targets the "fullscreen" button, hmm. 

> the gist of the general guidance is to use ARIA sparingly / as a last resort

Ahh, I did not know that, good to know!

**Comment by aminah-sc** [comment]
> I should have been more clear, for now we are looking solely at the content of the Home page, i.e. what is between the `<main>` tags.

No worries, this was clear (although I took it as including rather than between, but that's irrelevant with respect to the main point). It's for this reason that I started the FF inspector list at `<main>` rather than the preceding issues it highlights. My point is that I don't think you can fix the a11y issues with the homepage in `home-page.js`. 

The issue highlighted above about `<main>` also applies to the "What's here", "Where to begin" and other articles. For all of them the inspector says:

> Clickable elements must be focusable and should have interactive semantics.

Thing is, the elements highlighted (images in all cases, I believe), aren't actually clickable items, so why does the inspector think they are? I don't believe the answer is found in `home-page.js`.  

> It doesn't really work like that. The video is purely using the native player via the <video> element. IA is merely serving the file. that's why, if you view it in different browsers/platforms it appears quite different.

Yeah, in all honesty I don't have a clue about exactly how these things work. I spent quite a while trying to figure out what on earth was going on because I couldn't see any cause in `home-page.js` why NVDA and the inspector was throwing a fit of errors around the video player. Eventually I came to the article linked above, and then went to inspect the video directly on IA. I got the similar errors which is what led to my conclusion, although, NVDA's element list for each page is quite different. 

>  I don't see phantom focus points when tabbing: tabbing goes through the buttons, which I think is correct, except it omits the video.

When landing on the homepage, starting from the address bar it takes 26 tabs to go through the page (now that the sidebar is retracted by default, this is actually a great blessing by contrast to what it was before). But where exactly do each of these tabs take us (excluding the nav menu and video — which are inaccessible anyway — there are only 9 buttons on the page)? Here's what NVDA tells us for each, (it's only 9-26, in between `<main>`, that's of interest here; numbers and parentheses are my addition):

```
1. Menu button
2. Search SuttaCentral button
3. Search edit
4. clickable Menu button
5. clickable staging.suttacentral.net, visited link (SC logo)
6. SuttaCentral, heading 1, visited link
7. Early Buddhist texts, translations, and parallels, visited link
8. Tab control, introduction, introduction tab selected, 1 of 4.
9. Main, landmark clickable, main landmark, read this sutta, visited link
10. (nothing)
11. Introduction, button, link,
12. (nothing)
13. Get started, button, link,
14. (nothing)
15. More, button, link,
16. (nothing)
17. More, button, link,
18. (nothing)
19. More, button, link,
20. (nothing)
21. Testimonies, button, link,
22. (nothing)
23. Guides, button, link,
24. (nothing)
25. Indexes, button, link,
26. (nothing)
```
> in Chrome, but in FF it only targets the "fullscreen" button, hmm.

I can highly recommend this article: https://www.levelaccess.com/how-browsers-interact-with-screen-readers-and-where-aria-fits-in-the-mix. I think it was at the point of coming across this that Karl and I recognised getting things totally right a11y-wise wasn't a sensible way to think about things. Like all things, a11y involves a whole lot more mess and Grr than advocates (of which I am one) want to make out. 

Anyway, in short: obviously, I don't know js, but as far as I can tell, the homepage's key a11y problems are not rooted in `home-page.js`. It would, nevertheless, probably be good to put aria labels for the "more" buttons (ie, "Read more about the discourses" etc.)

**Comment by sujato** [comment]
closing this, only random quote remains and it has its own issue.
